scaleway/data_source_iam_group_test.go

```diff
@@ -0,0 +1,101 @@
+package scaleway
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccScalewayDataSourceIamGroup_Basic(t *testing.T) {
+	SkipBetaTest(t)
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testAccCheckScalewayIamGroupDestroy(tt),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_iam_group" "main_ds_basic" {
+						name        = "test_data_source_basic"
+					}
+			
+					data "scaleway_iam_group" "find_by_id_basic" {
+						group_id 	= scaleway_iam_group.main_ds_basic.id
+						organization_id = "08555df8-bb26-43bc-b749-1b98c5d02343"
+					}
+
+					data "scaleway_iam_group" "find_by_name_basic" {
+						name        = scaleway_iam_group.main_ds_basic.name
+						organization_id = "08555df8-bb26-43bc-b749-1b98c5d02343"
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayIamGroupExists(tt, "scaleway_iam_group.main_ds_basic"),
+					resource.TestCheckResourceAttr("data.scaleway_iam_group.find_by_id_basic", "name", "test_data_source_basic"),
+					resource.TestCheckResourceAttr("data.scaleway_iam_group.find_by_name_basic", "name", "test_data_source_basic"),
+					resource.TestCheckResourceAttrPair("data.scaleway_iam_group.find_by_id_basic", "id", "scaleway_iam_group.main_ds_basic", "id"),
+					resource.TestCheckResourceAttrPair("data.scaleway_iam_group.find_by_name_basic", "id", "scaleway_iam_group.main_ds_basic", "id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScalewayDataSourceIamGroup_UsersAndApplications(t *testing.T) {
+	SkipBetaTest(t)
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testAccCheckScalewayIamGroupDestroy(tt),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_iam_application" "app00" {
+						name = "app"
+					}
+					resource "scaleway_iam_group" "main_ds_mix" {
+						name = "test_data_source_mix"
+						application_ids = [
+							scaleway_iam_application.app00.id,
+						]
+						user_ids = [
```

Comment:
Can those user_ids be created using a terraform ressource to avoid having a hard-coded UUID?